service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/RemoveOrderControllerAboutToSubmitTest.java

```diff
@@ -160,6 +168,45 @@ void shouldNotUpdateChildrenPropertiesWhenRemovingANonFinalOrder() {
         assertThat(returnedChildren).isEqualTo(childrenList);
     }
 
+    @Test
+    void shouldRemoveCaseManagementOrderAndRemoveHearingAssociation() {
+        UUID removedOrderId = UUID.randomUUID();
+
+        Element<CaseManagementOrder> caseManagementOrder1 = element(removedOrderId, CaseManagementOrder.builder()
+            .status(APPROVED)
+            .build());
+
+        List<Element<CaseManagementOrder>> caseManagementOrders = List.of(
+            caseManagementOrder1,
+            element(CaseManagementOrder.builder().build()));
+
+        List<Element<HearingBooking>> hearingBookings = List.of(
+            element(HearingBooking.builder()
+                .caseManagementOrderId(removedOrderId)
+                .build()));
+
+        CaseData caseData = CaseData.builder()
+            .sealedCMOs(caseManagementOrders)
+            .hearingDetails(hearingBookings)
+            .removableOrderList(DynamicList.builder()
+                .value(buildListElement(removedOrderId, "Case management order - 15 June 2020"))
+                .build())
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse response = postAboutToSubmitEvent(caseData);
+
+        CaseData responseData = mapper.convertValue(response.getData(), CaseData.class);
```

Comment:
By having extended `AbstractControllerTest` you have access to the `extractCaseData` method